dist/vl-cookie-consent.js

```diff
@@ -329,7 +329,7 @@ export class VlCookieConsent extends vlElement(HTMLElement) {
   }
 
   _setCookie(name, value) {
-    document.cookie = this._cookiePrefix + name + '=' + value + ';Max-Age=2147483647;path=/;';
+    document.cookie = this._cookiePrefix + name + '=' + value + ';Max-Age=2147483647;path=/;SameSite=Strict;Secure';
```

Comment:
Zijn we zeker dat we hier "Strict" willen toepassen (en niet Lax). Als ik de specs correct interpreteer, worden dergelijke cookies niet meegegeven wanneer je vb vanuit een 3de applicatie komt.

Dus als ik denk aan ons scenario, waar een gebruiker klikt op een link die op de website van departement staat, zal dit betekenen dat de cookie niet wordt doorgegeven. Dus gebruiker zal altijd consent popup zien?

Best eens te testen of mijn interpretatie correct is...

Reply:
Secure is OK denk ik. Enkel wanneer je rechtstreeks op een node inlogt (dus zonder https) worden die cookies niet meegegeven, maar is enkel toegankelijk voor developers, eindgebruikers gaan altijd over https. (Niet getest maar misschien krijg je wel bij iedere navigatie een popup)

Reply:
Ik had al navraag gedaan bij @ldeschuyter of `Strict` of 'Lax` de voorkeur heeft, voorlopig heb ik het strengste gekozen. De cookie-consent moet sowieso per applicatie getoond worden, dus ik veronderstel dat `Strict` niet voor problemen zorgt.

Secure lijkt me ook niet voor problemen te zorgen, dus heb ik zoal gevraagd geactiveerd. Als het storend is voor developers, kan het nog altijd afgezet worden. Ook hier mag @ldeschuyter en in uitbreiding eventueel TAW hier feedback op geven.

Reply:
Klopt dat die per applicatie moet. Maar ik vermoed (niet getest) dat iedere keer je nu in de applicatie komt via een externe link, je die consent gaat zien. Dus er wordt geen rekening gehouden met vorige keuze... Enkel wanneer je rechtstreeks de URL ingeeft, zal die cookie van toepassing zijn.

Reply:
Te controleren of je vermoeden klopt. Ik had het vermoeden dat dit alleen relevant zou zijn voor asynchrone HTTP requests, maar heb het ook niet getest.

Reply:
Wie gaat controleren? :-)

Reply:
Ik denk dat jullie misschien best op ontwikkel een applicatie kunnen uitrollen die (tijdelijk) gebruik maakt van de branch en via een locale server kunnen jullie en/of ik dan testen wat het gedrag is. Lokaal is het moeilijker om een setup te maken met 2 verschillende hosts die naar elkaar linken.

Reply:
Mijn interpretatie was niet volledig juist. De browser stuurt de cookies inderdaad niet door *naar de server* de 1ste keer indien van je elders komt (hier zat ik juist), maar alles van die consent is client side  (dus hier ging ik de mist in)... Dus de frontend laadt en ze worden uitgelezen via JS.

Dus alles is OK.

*Strict* zou enkel problematisch zijn als iemand ook server side zou checken of de consent al dan niet werd geaccepteerd. Voorlopig is die use case er niet dus kan het zo blijven.

Reply:
Ik heb de strict verwijderd in kader van de unit/e2e testen die via http draaien. Met strict zouden de cookies niet geautomatiseerd kunnen worden met de huidige setup. Het lijkt me gezien bovenstaande communicatie geen probleem om strict niet toe te passen, anders verneem ik het wel.

Reply:
En wat met Lax ?

Reply:
Excuses, strict werd behouden maar secure werd niet toegepast.

Reply:
OK. Misschien te bekijken naar toekomst om (end2end) testen via HTTPS te draaien. Uiteindelijk is dit wat gebruikers ook doen.